py/dynruntime.h

```diff
@@ -140,9 +140,7 @@ static inline mp_obj_t mp_obj_cast_to_native_base_dyn(mp_obj_t self_in, mp_const
 
     if (MP_OBJ_FROM_PTR(self_type) == native_type) {
         return self_in;
-    }
-    mp_parent_t parent = mp_type_get_parent_slot(self_type);
-    if (parent != native_type) {
```

Comment:
this is probably correct for us. It looks like this is an incorrect trace of my work to split type objects into a main part and an optional part; the 'parent' field ended up in the main part, but this code was to help adapt for the case that it was in the optional part.